fx/vmap.py

```diff
@@ -0,0 +1,197 @@
+# This example is provided only for explanatory and educational purposes.
+import torch
+import torch.nn as nn
+import torch.fx as fx
+from torch.fx import Proxy
+from typing import Tuple, Any, Optional
+
+import torch.fx
+from torch.fx.passes.shape_prop import ShapeProp
+
+# Vmap
+# ---------------
+# `vmap` (short for vectorizing map) is a function transformation that takes in
+# a model that operates on single examples and returns a model that operates on
+# multiple examples. For example, if our model `M` originally takes in tensors
+# of shape (H, W) and returns a scalar, then `vmap(M)` should take in tensors
+# of shape `(B, H, W)` and return a vector. This procedure is also often called
+# "batching" a model.
+#
+# How is this feat accomplished? One observation is that to "batch" a model, it
+# suffices to batch each individual operation. In other words, given an
+# operation that works on the current shape, how do we make it work on another
+# batch dimension? This leads us to batching rules.
```

Comment:
s/on another batch dimension/with an additional batch dimension/

I think that reads a little clearer